src/harness/unittests/tscWatchMode.ts

```diff
@@ -126,1900 +105,1334 @@ namespace ts.tscWatch {
-        });
+            });
 
-        it("Configure file diagnostics events are generated when the config file has errors", () => {
-            const file = {
-                path: "/a/b/app.ts",
-                content: "let x = 10"
-            };
-            const configFile = {
-                path: "/a/b/tsconfig.json",
-                content: `{
-                        "compilerOptions": {
-                            "foo": "bar",
-                            "allowJS": true
-                        }
-                    }`
-            };
-
-            const host = createWatchedSystem([file, configFile, libFile]);
-            const watch = createWatchModeWithConfigFile(configFile.path, host);
-            checkOutputErrors(host, [
-                getUnknownCompilerOption(watch(), configFile, "foo"),
-                getUnknownCompilerOption(watch(), configFile, "allowJS")
+            it("Configure file diagnostics events are generated when the config file has errors", () => {
+                const configFileContent = `{ "compilerOptions": { "foo": "bar", "allowJS": true } }`;
+                const host = new fakes.FakeServerHost({ lib: true });
+                host.vfs.addFile("/a/b/app.ts", `let x = 10`);
+                host.vfs.addFile("/a/b/tsconfig.json", configFileContent);
+
+                const watch = createWatchModeWithConfigFile("/a/b/tsconfig.json", host);
+                checkOutputErrors(host, [
+                    createUnknownCompilerOptionDiagnostic(watch(), configFileContent, "foo"),
+                    createUnknownCompilerOptionDiagnostic(watch(), configFileContent, "allowJS")
             ], /*errorsPosition*/ ExpectedOutputErrorsPosition.BeforeCompilationStarts);
-        });
+            });
+
+            it("If config file doesnt have errors, they are not reported", () => {
+                const host = new fakes.FakeServerHost({ lib: true });
+                host.vfs.addFile("/a/b/app.ts", `let x = 10`);
+                host.vfs.addFile("/a/b/tsconfig.json", `{ "compilerOptions": {} }`);
 
-        it("If config file doesnt have errors, they are not reported", () => {
-            const file = {
-                path: "/a/b/app.ts",
-                content: "let x = 10"
-            };
-            const configFile = {
-                path: "/a/b/tsconfig.json",
-                content: `{
-                        "compilerOptions": {}
-                    }`
-            };
-
-            const host = createWatchedSystem([file, configFile, libFile]);
-            createWatchModeWithConfigFile(configFile.path, host);
+                createWatchModeWithConfigFile("/a/b/tsconfig.json", host);
             checkOutputErrors(host, emptyArray, /*errorsPosition*/ ExpectedOutputErrorsPosition.AfterCompilationStarting);
-        });
+            });
+
+            it("Repor
```

Comment:
I still dislike seeing `_ => _` repeated here, even though I've already been informed that it's here because it's the only way to ensure the call is strongly typed (since you can't get a tuple type from a function's argument list or similar). I guess I'm fine enough with it, I just wanted to document that it makes me sad.